core/src/main/scala/uk/gov/nationalarchives/tdr/api/core/graphql/service/FileService.scala

```diff
@@ -35,10 +35,11 @@ class FileService(fileDao: FileDao, consignmentService: ConsignmentService)(impl
     val newFile = FileRow(None, path, consignmentId)
     val result = fileDao.create(newFile)
 
-    result.flatMap(persistedFile =>
+    result.flatMap(persistedFile => {
+      fileStatusService.create(persistedFile.id.get)
```

Comment:
This returns a future, so we should map it rather than just going straight to getting the consignment on the next line.

Reply:
I've replaced this with the for expression which has also tidied it up a bit.